### ISSUE TITLE
Buy Button

### ISSUE THREAD

**Issue by seekthelight2** [nbrightproject/NBrightBuy#46]
I have products with multiple options where the buy button seems no to operate. It won't let me add to the cart. mi1Printing.com

**Comment by leedavi** [comment]
Do you see any errors in the event log?

It's also odd that it allows you to add to basket from the product list, this shouldn't happen when you have product options, but that might just be something missing from the display template.

If you send me an export of your data I'll give it a try and my dev machine.

**Comment by seekthelight2** [comment]
[mi1printing.com_export.zip](https://github.com/nbrightproject/NBrightBuy/files/465213/mi1printing.com_export.zip)

So, in the event viewer there are just to scheduler issues with the store and that's it. Just FYI, I'm using DNN 8.0.3 and NBStore 3.5

#1 THREAD ID:17
TYPE:Nevoweb.DNN.NBrightBuy.Components.NBrightBuyScheduler,NBrightBuy
EXCEPTION:Object reference not set to an instance of an object.
RESCHEDULED FOR:9/9/2016 11:09:04 PM
SOURCE:STARTED_FROM_TIMER
ACTIVE THREADS:0
FREE THREADS:1
READER TIMEOUTS:0
WRITER TIMEOUTS:0
IN PROGRESS:1
IN QUEUE:6

and #2 AbsoluteURL:
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:0ff8c3ca-647f-4e1f-ac68-c9ad60b7eb40
AssemblyVersion:
PortalId:-1
UserId:-1
TabId:-1
RawUrl:
Referrer:
UserAgent:
ExceptionHash:zq8i0q+EEdR+BgrgaMmQsg==
Message:Object reference not set to an instance of an object.
StackTrace:
   at Nevoweb.DNN.NBrightBuy.Components.PluginData.CalcPluginList(NBrightInfo info)
   at Nevoweb.DNN.NBrightBuy.Components.PluginData.Load()
   at Nevoweb.DNN.NBrightBuy.Components.PluginData..ctor(Int32 portalId, Boolean systemlevel)
   at Nevoweb.DNN.NBrightBuy.Components.NBrightBuyScheduler.DoWork()
InnerMessage:
InnerStackTrace:
Source:NBrightBuy
FileName:
FileLineNumber:0
FileColumnNumber:0
Method:

**Comment by seekthelight2** [comment]
Any ideas?

**Comment by leedavi** [comment]
OK, sorry, this got lost in the pile.

the problem is the inches sign on the option dropdown:  Size: 3.5" X 2"

It's causing a problme with the XML rendering. I'll put a fix in the next version.

If you need to get it working with the current code you'll need to remove the inches sign : Size: 3.5 X 2

**Comment by seekthelight2** [comment]
Thank you. That solves this issue. While I have you, however, when the "buy" button is pressed the item is added to the cart but nothing else happens. The screen doesn't change, wording doesn't pop up, like "added to cart". Did I forget a setting somewhere, otherwise, how would the customer know the item was added to the cart?

**Comment by leedavi** [comment]
There should be a line in the NBS_ProductDisplayDetail.cshtml that popups on click add to basket.

<div class="addedtobasket addbasketmsg@(info.ItemID)" style="display:none;"><span>@ResourceKey("ProductView.addedtobasket")</span></div>

I know in 1 version it wasn;t working on the detail, sounds like your running that version.

make sure you have the minibasket on the same page, the code to make that popup is in the minibasket templates.

**Comment by leedavi** [comment]
```
    $('.addtobasket').click(function () {
        //if (form.valid()) {
            nbxget('addtobasket', '.entryid' + $(this).attr('itemid'), '#minicartdatareturn'); // Reload Cart
            $('.addedtobasket').delay(10).fadeIn('fast');
        //}
    });

    $('#minicartdatareturn').change(function () {
        nbxget('rendercart', '.minicartdata', '.cartpane'); // Reload Cart
        $('.addedtobasket').delay(1000).fadeOut('fast');
    });
```

**Comment by seekthelight2** [comment]
I found out that it does have a light blue message that reads, "Item Added to Cart", however it appears at the top of the page and if you don't scroll up quickly, you won't see it. Is it possible to make the message appear near the buy button?

**Comment by leedavi** [comment]
This might be just a css issue.  First try downloading the latest version of ClassicRazor  (You'll need to download the whole zip project I think), 

https://github.com/nbrightproject/NBrightBuy/tree/master/Themes/ClassicRazor

You should just be able to replace the current folder with the one from GitHub.  Then restart the app pool.

**Comment by Tychodewaard** [comment]
The latest version doesn't fix this. I would recommend a UX like http://demo.nopcommerce.com/nikon-d5500-dslr (tap/click add to cart). This way on every device / resolution the feedback is clear. @leedavi : If sponsoring is needed to get it higher on the list, let me know.  I asked Geoff the same question. If he has some time left, I'll get the code to you.

**Comment by in8dna** [comment]
For others.. The styling is at line 245 as .addedtobasket span in the classicrazor.css file. We tweaked it into a much larger notification:
`.addedtobasket span {background:#2ECC71;color:white;-moz-border-radius:7px;-webkit-border-radius:7px;border-radius:7px;padding:15px 24px;font-size:18px;}`

You'll also find the 'Flag' duration (delay) setting in the NBS_MiniCart_Head.cshtml file line 61

`$('#minicartdatareturn').change(function () {
            nbxget('rendercart', '.minicartdata', '.container_classicrazor_nbs_minicart'); // Reload Cart
            $('.addedtobasket').delay(3000).fadeOut('slow');`

I agree with Tychodewaard that the UX he points to is clear and works across devices nicely.

**Comment by Tychodewaard** [comment]
Nice on, @in8dna. I'm gonna play with that one.

I also took a look at the nopcommerce solution: 
As a non-dev I'm not able to fix this myself. But I saw the concept (screen attached): default the green bar is display=none. The addedtoBasket function removes the display=none which makes it visible. 

![schermafbeelding 2016-10-05 om 20 12 52](https://cloud.githubusercontent.com/assets/9091997/19125669/429f0fe0-8b38-11e6-8284-18a1392b1c1b.png)
